### PR TITLE
Bug/save unsave feature

### PR DESCRIPTION
**Pull request by SrinandhiniM**
Bugs posted in Save and Unsave are corrected now.

### CODE REVIEW

File: app/controllers/public_controller.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ def index
   end
   def profile
     community_list
-    
+    @saved_posts = SavePost.where(account_id: current_account.id)
```
Comment (iMagesh) [inline app/controllers/public_controller.rb:11]: the SavePost query seems to be duplicated in two places, can you move it to a common place and access it in all the methods instead of creating a duplicate? 
Follow DRY everywhere

File: app/controllers/public_controller.rb [hunk 2]
```diff
@@ -17,6 +17,10 @@ def profile
     @my_comments = Comment.where(account_id: current_account.id)
   end
 
+  def saved_posts
+    @saved_posts = SavePost.where(account_id: current_account.id)
```
Comment (iMagesh) [inline app/controllers/public_controller.rb:21]: See you have used it here

File: app/javascript/packs/application.js [hunk 3]
```diff
@@ -17,7 +17,6 @@ require("../includes/infinitescroll.js")
 require("../includes/navbar.js")
 require("../includes/discussion.js")
 require("../includes/comment_form.js")
-require("../includes/autocomplete.js")
```
Comment (iMagesh) [inline app/javascript/packs/application.js:20]: why did you remove the autocomplete.js? is it your code or someone else's?

**Review by iMagesh — CHANGES_REQUESTED** [review]
check the comments and update your code

File: app/javascript/packs/application.js (hunk 3 above)
Comment (SrinandhiniM) [reply app/javascript/packs/application.js:20]: Auto complete js is not working. Because of that file, some Ajax functions didn't work. So removed that file

File: app/controllers/public_controller.rb (hunk 2 above)
Comment (SrinandhiniM) [reply app/controllers/public_controller.rb:21]: deleted that code now

File: config/routes.rb [hunk 4]
```diff
@@ -13,14 +13,23 @@
...
   end
+
+  resources :save_post do
+    member do
+      patch :save
+      patch :unsave
+    end
+  end
+
+  # get "/saved_posts" => "save_posts#saved_posts"
```
Comment (Venirails) [inline config/routes.rb:24]: remove the commented lines if they are no longer required

Comment (SrinandhiniM) [reply config/routes.rb]: Removed the commented lines

**Comment by SrinandhiniM** [comment]
Now all the corrections done

**Comment by iMagesh** [comment]
@SrinandhiniM please make the DB table-related changes we discussed and then let me know so we can merge this.

**Comment by SrinandhiniM** [comment]
Now migrations are corrected. We need account id and post id for reference, I deleted community id reference. Now all are working perfectly.

**Comment by iMagesh** [comment]
> Now migrations are corrected. We need account id and post id for reference, I deleted community id reference. Now all are working perfectly.

We don't need account_id since it is availble inside post. Remember? I asked you to do joins?

**Comment by SrinandhiniM** [comment]
We need current Acoount ID for saving the post. Post contains id account id of those who posted the post. But we need current ID to save the post .

File: app/controllers/save_post_controller.rb [hunk 5]
```diff
@@ -7,10 +7,10 @@ def create
   def save
     @saved_post = SavePost.new(
       account_id: current_account.id,
-      post_id: params[:id],
-      community_id: Post.find_by(id: params[:id]).community_id
+      post_id: params[:id]
+      #community_id: Post.find_by(id: params[:id]).community_id
```
Comment (iMagesh) [inline app/controllers/save_post_controller.rb:11]: remove the comments before committing code @SrinandhiniM

**Comment by SrinandhiniM** [comment]
Comments removed and pulled from Master and resolved conflicts.

**Comment by iMagesh** [comment]
Please don't delete the migration files if the tables are necessary. Schema files will be updated very often and so we can't rely on that. We need all the migration files to regenerate the schema whenever required. @SrinandhiniM

**Comment by SrinandhiniM** [comment]
Migrations files are created now correctly.
